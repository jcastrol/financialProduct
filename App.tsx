import React from 'react';
import {GestureHandlerRootView} from 'react-native-gesture-handler';
import {RootStack} from './src/routes/RootStack';
import {NavigationContainer} from '@react-navigation/native';
import {QueryClient, QueryClientProvider} from '@tanstack/react-query';

function App(): React.JSX.Element {
  const queryClient = new QueryClient();

  return (
    <GestureHandlerRootView style={{flex: 1}}>
      <QueryClientProvider client={queryClient}>
        <NavigationContainer>
          <RootStack />
        </NavigationContainer>
      </QueryClientProvider>
    </GestureHandlerRootView>
  );
}

export default App;
