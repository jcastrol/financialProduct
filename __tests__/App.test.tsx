import React from 'react';
import { render } from '@testing-library/react-native';
import App from '../App';


jest.mock('react-native-vector-icons/MaterialCommunityIcons', () => {
  return  jest.fn() 
});
jest.mock('react-native-vector-icons/FontAwesome6', () => {
  return  jest.fn() 
});
jest.mock('@react-navigation/native', () => {
  return {
    NavigationContainer: ({ children }: { children: React.ReactNode }) => children,
  };
});

jest.mock('@react-navigation/native-stack', () => {
  return {
    createNativeStackNavigator: jest.fn(() => {
      return {
        Navigator: ({ children }: { children: React.ReactNode }) => children,
        Screen: () => null,
      };
    }),
  };
});

jest.mock('react-native-gesture-handler', () => {
  return {
    GestureHandlerRootView: ({ children }: { children: React.ReactNode }) => children,
  };
});

jest.mock('@tanstack/react-query', () => {
  return {
    QueryClient: jest.fn(),
    QueryClientProvider: ({ children }: { children: React.ReactNode }) => children,
  };
});

describe('App', () => {
  it('renders correctly and matches snapshot', () => {
    const tree = render(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
