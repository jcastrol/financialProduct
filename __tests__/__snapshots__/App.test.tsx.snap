// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly and matches snapshot 1`] = `null`;
